### PR TITLE
DOCSP-29510: explain how driver handles undefined vals

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-node/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-29510
Staging:
[Landing page for new fundamentals section
](https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/DOCSP-29510-explain-undefined-values/fundamentals/bson/)
[Undefined values page](https://docs-mongodbcom-staging.corp.mongodb.com/node/docsworker-xlarge/DOCSP-29510-explain-undefined-values/fundamentals/bson/undefined-values/#std-label-node-undefined-values)

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/fundamentals/crud/write-operations/change-a-document.txt [hunk 1]
```diff
@@ -205,3 +205,5 @@ inserts a new document with the specified fields and values.
 You cannot modify the ``_id`` field of a document nor change a field to
 a value that violates a unique index constraint. See the MongoDB Server manual
 for more information on :manual:`unique indexes </core/index-unique/>`.
+
+.. include:: /includes/undefined-values-note.rst
```
Comment (rustagir) [inline source/fundamentals/crud/write-operations/change-a-document.txt:209]: I'm undecided on whether this was the right approach, as opposed to making different content for these pages based on the type of write op that they explain. I also couldn't find the right place to put this material, as it didn't seem appropriate to make a whole subsection for it on each page.

Comment (nbbeeken) [reply source/fundamentals/crud/write-operations/change-a-document.txt:209]: We have the UTF-8 Validation docs as a top level page but this and that UTF page are both BSON controls. Maybe there's a merging or nesting that would be sensical given that. 

I foresee a near future where we may be adding more coverage of the BSON controls that the driver offers, we just have to [make the options more easily understood first](https://jira.mongodb.org/browse/NODE-4780).

Just a suggestion, of course!

File: config/redirects [hunk 2]
```diff
@@ -17,3 +17,4 @@ raw: ${prefix}/stable -> ${base}/current/
 [*-v5.0]: ${prefix}/${version}/quick-start/download-and-install/ -> ${base}/${version}/quick-start/
 [*-v5.0]: ${prefix}/${version}/quick-start/next-steps/ -> ${base}/${version}/quick-start/
 
+[v4.3-master]: ${prefix}/${version}/fundamentals/utf8-validation -> ${base}/${version}/fundamentals/bson/utf8-validation/
```
Comment (rustagir) [inline config/redirects:20]: [mut redirect test](https://gist.github.com/rustagir/86546acc9c6a3d2bc5d8b3d7959970dc)

File: source/fundamentals/bson/undefined-values.txt [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+   const myColl = myDB.collection("states");
+
+   // Any write operation will not serialize undefined values
+   await myColl.findOneAndUpdate(
+     { state: "Georgia" },
+     { $set: { gasTax: undefined } }
+   );
+
+You can override the setting at any level of scope by including it when
+constructing the object instance or when executing an operation.
```
Comment (jordan-smith721) [inline source/fundamentals/bson/undefined-values.txt:99]: I found this a bit hard to parse out. It could possibly be simplified to something like this, since there is an example immediately following.

```suggestion
You can specify ``ingoreUndefined` again at any level to override the inherited settings.
```

File: source/fundamentals/bson/undefined-values.txt [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+Overview
+--------
+
+In this guide, you can learn how to control how the driver serializes
+undefined values. By default, the driver serializes ``undefined`` values
+as ``null`` values during write operations.
+
+.. _nodejs-specify-ignoreundefined:
+
+Specify the ``ignoreUndefined`` Setting
```
Comment (jordan-smith721) [inline source/fundamentals/bson/undefined-values.txt:22]: Suggestion: I think this section could potentially be renamed something like "Ignore Undefined Values" since that's the core of what they are learning how to do here.

**Review by jordan-smith721 — CHANGES_REQUESTED** [review]
Nice work on this! I've got a few suggestions that I think could make the page a bit easier to digest with how technical it is.

File: source/fundamentals/bson/undefined-values.txt [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+   );
+
+You can override the setting at any level of scope by including it when
+constructing the object instance or when executing an operation.
+
+For example, if you set ``ignoreUndefined`` to ``true`` on your
+collection object, you can override the setting in individual write
+operations that you execute on that collection.
+
+.. code-block:: javascript
```
[the remark this answers is not in the document]
Comment (rustagir) [reply source/fundamentals/bson/undefined-values.txt:105]: technically, this update operation would not cause any changes to the data. I can include that somewhere in the text, if you think it would be useful for users to know

File: source/fundamentals/bson/undefined-values.txt [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+To make the driver ignore fields with
+``undefined`` values during serialization, set the
+``ignoreUndefined`` setting to ``true``. When you specify this setting,
+the driver *does not* serialize fields with ``undefined`` values.
+
+The following example inserts two documents. One insert operation has
+the ``ignoreUndefined`` setting set to ``true``. The driver does not
+serialize the ``salesTax`` field during the first insert operation
+because we specified the ``ignoreUndefined`` setting:
```
Comment (jordan-smith721) [inline source/fundamentals/bson/undefined-values.txt:33]: Suggestion: I think these sentences can be combined:

```suggestion
The following example inserts two documents. The first insert operation has
the ``ignoreUndefined`` setting set to ``true``, so it does not
serialize the ``salesTax`` field.
```

**Review by jordan-smith721 — APPROVED** [review]
Nice work with the second pass! LGTM with one small suggestion
